vignettes/ees-api-workflow.Rmd

```diff
@@ -0,0 +1,282 @@
+
+Using the parameters we obtained from `get-meta` above, then we can create a query to return 
+session count data for only **Approved educational activity**, **Authorised** and **Unauthorised** 
+in the **Absences** category, **Nationally** for **York** in **Week 26** in **2024**, we would 
+supply the following:
+
+``` {r example-parameter-query}
+eesyapi::query_dataset(
+  dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6",
+  indicators = "bqZtT",
+  filter_items = list(
+    attendance_status = c("7SdXo"),
+    attendance_type = c("6AXrf", "0k3T5", "YdkHK"),
+    day_number = "uLQo4",
+    reason = "bBrtT"
+  ),
+  geographies = c("NAT|id|dP0Zw", "LA|ID|jfhAM"),
+  time_periods = c("2024|W26")
+)
+```
+
+Some notes on the above:
+
+- `indicators`, `time_periods`, `geographies` and `filter_items` can all be supplied as vectors - 
+`c(...)` - and will be interpreted as `filter(... %in% c(...))`.
+- `filter_items` can be supplied as a list to create more stringent cross-filter combinations, i.e. 
+in the above example rows will only be returned for the specific combinations given across the 4 
+filters.
+- `geographies` can be supplied as a data frame to allow for more complex queries such as "give me 
+all LAs in a given region". The structure for this can be seen by calling `example_geography_query()`.
+
+
+#### Querying a data set using a json file or string
+
+If you're familiar with using json and wish to just write your own query from scratch, then you can 
+use `query_dataset()` with the `json_query` parameter as follows:
+
+``` {r example-json-string-query}
+json_query <- "{
+  \"criteria\": {
+    \"and\": [
+      {
+        \"filters\": {
+          \"eq\": \"7SdXo\"
+        }
+      },
+      {
+        \"locations\": {
+          \"eq\": {
+            \"level\": \"LA\",
+            \"id\": \"jfhAM\"
+          }
+        }
+      }
+    ]
+  },
+  \"indicators\": [
+    \"bqZtT\"
+  ],
+  \"debug\": false,
+  \"page\": 1,
+  \"pageSize\": 10
+}"
+eesyapi::query_dataset(
+  dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6",
+  json_query = json_query
+)
+```
+
+## FAQs
+
+### Why do you use IDs rather than the actual labels used in the data?
+
+The IDs are intended to aid in future proofing fir users that may develop automated pipelines that
+connect to the EES API. Labeling inevitably changes, for example local authorities sometimes get
+renamed or column labels are updated to be clearer. If the underlying content of that item stays
+the same, i.e. the renamed local authority still has the same boundaries, then automated pipelines
+will be required to still connect to the same item, therefore each item has an associated ID that
+will remain unchanged in a given data set as it is updated with new data over the subsequent
+releases, even if a renaming of a field occurs.
+
+### Are there any bandwidth / repeat connection constraints that I should know about?
+
+...
```

Comment:
Nothing currently while we're in testing

Reply:
Brief bit of text added here.